core/src/main/scala/org/apache/spark/rpc/RpcEnv.scala

```diff
@@ -187,7 +188,10 @@ private[spark] trait RpcEnvFileServer {
 
   /** Validates and normalizes the base URI for directories. */
   protected def validateDirectoryUri(baseUri: String): String = {
-    val baseCanonicalUri = new File(baseUri).getCanonicalPath
+    var baseCanonicalUri = new File(baseUri).getCanonicalPath
```

Comment:
Can we use URI(...).getParts instead?

Reply:
Same comment from https://github.com/apache/spark/pull/36089#discussion_r845688822

Reply:
Sorry, the last PR was accidentally closed. Thank you very much for your timely reply

`var baseCanonicalUri = new File(baseUri).getCanonicalPath `

This line is added by [spark-35691]. Why is it so modified? You can check [spark-35691]. If we use  URI(...).getParts   instead, we will lose the improvement of  [spark-35691]   . This is my understanding. Please correct any mistakes